MODULE.bazel

```diff
@@ -15,13 +15,39 @@ module(
 bazel_dep(name = "abseil-cpp", version = "20230802.0.bcr.1", repo_name = "com_google_absl")
 bazel_dep(name = "bazel_skylib", version = "1.4.1")
 bazel_dep(name = "jsoncpp", version = "1.9.5")
+bazel_dep(name = "platforms", version = "0.0.8")
 bazel_dep(name = "rules_cc", version = "0.0.9")
-bazel_dep(name = "rules_java", version = "5.3.5")
-bazel_dep(name = "rules_jvm_external", version = "5.1")
+bazel_dep(name = "rules_java", version = "6.0.0")
+bazel_dep(name = "rules_jvm_external", version = "6.0")
 bazel_dep(name = "rules_pkg", version = "0.7.0")
 bazel_dep(name = "rules_python", version = "0.10.2")
-bazel_dep(name = "platforms", version = "0.0.8")
 bazel_dep(name = "zlib", version = "1.2.11")
 
 # TODO: remove after toolchain types are moved to protobuf
-bazel_dep(name = "rules_proto", version = "4.0.0")
\ No newline at end of file
+bazel_dep(name = "rules_proto", version = "4.0.0")
+
+maven = use_extension("@rules_jvm_external//:extensions.bzl", "maven")
+maven.install(
+    artifacts = [
+        "com.google.caliper:caliper:1.0-beta-3",
+        "com.google.code.findbugs:jsr305:3.0.2",
+        "com.google.code.gson:gson:2.8.9",
+        "com.google.errorprone:error_prone_annotations:2.18.0",
+        "com.google.j2objc:j2objc-annotations:2.8",
+        "com.google.guava:guava:32.0.1-jre",
+        "com.google.guava:guava-testlib:32.0.1-jre",
+        "com.google.truth:truth:1.1.2",
+        "junit:junit:4.13.2",
+        "org.mockito:mockito-core:4.3.1",
```

Comment:
This is problematic: All these deps leak into the default dependency closure. Only actual runtime deps should.

Reply:
are these really just dev deps? https://github.com/protocolbuffers/protobuf/blob/ee98ba2c18486938952a7748765f2e8e119e7fc4/java/util/BUILD.bazel#L16

Reply:
https://github.com/protocolbuffers/protobuf/blob/ee98ba2c18486938952a7748765f2e8e119e7fc4/BUILD.bazel#L353-L357

when i try to integrate w/ this patch into grpc-java this target is used which requires protobuf have a `@maven` with this stuff

Reply:
I think that we need to split this into a dev and a prod list of maven deps, with two different hub repo names

Reply:
I just found another repo where you needed this in the WORKSPACE too so yea I think that's just an issue that folks are probably avoiding by not updating protobuf